src/Turing.jl

```diff
@@ -68,6 +56,13 @@ using .ModeEstimation
     export optimize
 end
 
+if !isdefined(Base, :get_extension)
+    using Requires
+end
+@static if !isdefined(Base, :get_extension)
+    @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" include("../ext/DynamicHMCExt.jl")
```

Comment:
Sorry if my previous suggestion was misleading: This line should be inside of `___init___` - but the `@static` check should be wrap `__init___`. So if extensions are supported, we don't define a custom __init__ function at all.

Reply:
ok but if I try `@static function __init__()` it won't compile.
Is there something I am missing? Sorry for being dense.

Reply:
It's a pattern used all over the Julia ecosystem, so in prinicple it has to work. The problem here, however, might be the additional `@init`s - they also change the `__init__` function and I don't know how this interacts with this approach.